### PR TITLE
add coding standards

### PR DESCRIPTION
**Pull request by qingyang-id**
### Description
add coding standards for contract and js files

### Rationale

Let every one follow the same rule, The code is highly readable and reduce code conflicts, some errors can be found in advance and etc.
Currently it is not mandatory to strictly follow the coding standard, you can turn off the eslint plugin to ignore it, but the coding standard will be enforced in the future.

### Example

yarn check : check whether the code follows the coding standard
yarn fix : fix the code style, let them follow the coding standard

### Changes
add eslint, prettier, solhint and related packages
add eslint, prettier, solhint rule config and scrpits

Notable changes:
* This time the code style is not fixed. If you enable the eslint plugin, you will see a lot of reminders because they do not follow the code standards, but this does not affect the code running. If you are the original author, you can help to fix the coding style, I think the fix will not have any impact on the code logic

### CODE REVIEW

**Review by rzrrbk — COMMENTED** [review]
LGTM 🔥

**Review by JonneyTheSheep — COMMENTED** [review]
/lgtm

File: package.json [hunk 1]
```diff
@@ -4,7 +4,18 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "build": "hardhat compile --config ./hardhat.config.js",
```
Comment (srimannem) [inline package.json]: No need to add the --config flag. Its already on the root directory

File: package.json [hunk 2]
```diff
@@ -4,7 +4,18 @@
...
+    "test": "hardhat test --config ./hardhat.config.js",
+    "profile": "REPORT_GAS=true hardhat test --config ./hardhat.config.js",
+    "lint:check:format": "prettier --check **.{sol,js}",
+    "lint:check:solhint": "solhint contracts/**/*.sol",
+    "lint:check:eslint": "eslint . --ext js,ts",
+    "check": "yarn lint:check:format && yarn lint:check:solhint && yarn lint:check:eslint",
+    "lint:fix:format": "prettier --write **.{sol,js}",
+    "lint:fix:eslint": "eslint --fix . --ext js",
+    "fix": "yarn lint:fix:format && yarn lint:fix:eslint",
+    "verify": "npx hardhat verify"
```
Comment (srimannem) [inline package.json]: verify needs arguments. This is better called from the terminal using npx commands. We can remove it from here

File: package.json [hunk 3]
```diff
@@ -4,7 +4,17 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (deeprnd) [inline package.json:7]: perhaps better leave it as a reminder to integrate tests later

**Review by lightning-li — APPROVED** [review]
LGTM

**Pull request merged**